fdbserver/include/fdbserver/workloads/TenantConsistency.actor.h

```diff
@@ -36,222 +36,120 @@
 #include "fdbclient/MetaclusterManagement.actor.h"
 #include "fdbclient/Tenant.h"
 #include "fdbclient/TenantManagement.actor.h"
+#include "fdbserver/workloads/TenantData.actor.h"
 #include "flow/actorcompiler.h" // This must be the last #include.
 
-template <class DB>
+template <class DB, class TenantMapEntryT>
 class TenantConsistencyCheck {
 private:
-	Reference<DB> db;
-
-	struct TenantData {
-		Optional<MetaclusterRegistrationEntry> metaclusterRegistration;
-		std::map<TenantName, int64_t> tenantNameIndex;
-		int64_t lastTenantId;
-		int64_t tenantCount;
-		std::set<int64_t> tenantTombstones;
-		Optional<TenantTombstoneCleanupData> tombstoneCleanupData;
-		std::map<TenantGroupName, TenantGroupEntry> tenantGroupMap;
-		std::map<TenantGroupName, std::set<int64_t>> tenantGroupIndex;
-
-		std::set<int64_t> tenantsInTenantGroupIndex;
-
-		ClusterType clusterType;
-	};
-
-	TenantData metadata;
+	TenantData<DB, TenantMapEntryT> tenantData;
 
 	// Note: this check can only be run on metaclusters with a reasonable number of tenants, as should be
 	// the case with the current metacluster simulation workloads
 	static inline const int metaclusterMaxTenants = 10e6;
 
-	ACTOR template <class TenantMapEntryImpl>
-	static Future<std::map<int64_t, TenantMapEntryImpl>> loadTenantMetadataImpl(
-	    TenantConsistencyCheck* self,
-	    TenantMetadataSpecification<TenantMapEntryImpl>* tenantMetadata,
-	    Reference<typename DB::TransactionT> tr) {
-		state KeyBackedRangeResult<std::pair<int64_t, TenantMapEntryImpl>> tenantList;
-		state KeyBackedRangeResult<std::pair<TenantName, int64_t>> tenantNameIndexList;
-		state KeyBackedRangeResult<int64_t> tenantTombstoneList;
-		state KeyBackedRangeResult<std::pair<TenantGroupName, TenantGroupEntry>> tenantGroupList;
-		state KeyBackedRangeResult<Tuple> tenantGroupTenantTuples;
-
-		loop {
-			try {
-				tr->setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
-				wait(
-				    store(tenantList, tenantMetadata->tenantMap.getRange(tr, {}, {}, metaclusterMaxTenants)) &&
-				    store(tenantNameIndexList,
-				          tenantMetadata->tenantNameIndex.getRange(tr, {}, {}, metaclusterMaxTenants)) &&
-				    store(self->metadata.lastTenantId, tenantMetadata->lastTenantId.getD(tr, Snapshot::False, -1)) &&
-				    store(self->metadata.tenantCount, tenantMetadata->tenantCount.getD(tr, Snapshot::False, 0)) &&
-				    store(tenantTombstoneList,
-				          tenantMetadata->tenantTombstones.getRange(tr, {}, {}, metaclusterMaxTenants)) &&
-				    store(self->metadata.tombstoneCleanupData, tenantMetadata->tombstoneCleanupData.get(tr)) &&
-				    store(tenantGroupTenantTuples,
-				          tenantMetadata->tenantGroupTenantIndex.getRange(tr, {}, {}, metaclusterMaxTenants)) &&
-				    store(tenantGroupList, tenantMetadata->tenantGroupMap.getRange(tr, {}, {}, metaclusterMaxTenants)));
+	void validateTenantMetadataImpl() {
```

Comment:
More of a note than a change request, but this will likely need to be specialized based on `TenantGroupEntry` versus `MetaclusterTenantGroupEntry` when it is split